### PR TITLE
Document how to wrap a field React component into Flow component

### LINKED ISSUE #3421: Document how to wrap a field React component into Flow component
**Tell us about your request**

Add a note and code example into [React integration docs](https://vaadin.com/docs/next/flow/integrations/react) how to wrap a React component as a Field Flow component with `AbstractSinglePropertyField` before a new connector API is added within https://github.com/vaadin/flow/issues/19276.

---

If you think this issue is important, add a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to help the community and maintainers prioritize this issue.

### PR DESCRIPTION
**Pull request by AlainaFaisal**
Document of an example in [React integration docs](https://vaadin.com/docs/next/flow/integrations/react) how to wrap a React component as a Field Flow component with AbstractSinglePropertyField

Fixes #3421

### CODE REVIEW

File: articles/flow/integrations/react.adoc [hunk 1]
```diff
@@ -208,3 +208,80 @@ class ReactRouterLayoutElement extends ReactAdapterElement {
...
+import com.vaadin.flow.component.dependency.JsModule;
+import com.vaadin.flow.component.textfield.AbstractSinglePropertyField;
+
+@Tag("react-text-input")
+@JsModule("./src/react-text-input.js")
+public class ReactTextField extends AbstractSinglePropertyField<ReactTextField, String> {
+public ReactTextField() {
+super("value", "", false);
+// Synchronize the "value" property on the React side with the Java side whenever the "value-changed" event is fired
+getElement().synchronizeProperty("value", "value-changed");
```
Comment (vursen) [inline articles/flow/integrations/react.adoc:271]: Just noticed that AbstractSinglePropertyField already sets up the property synchronization in its [constructor](https://github.com/vaadin/flow/blob/29e2e4adff260516c3d75789f52600d7e10543dd/flow-server/src/main/java/com/vaadin/flow/component/AbstractSinglePropertyField.java#L250-L251), so this part is probably unnecessary?

Comment (AlainaFaisal) [reply articles/flow/integrations/react.adoc:271]: @vursen Thank you for your comment! I had added this as 'value-changed', in this case, was a custom event emitted by the React component to communicate changes back to the Vaadin server component. And I thought that the default behavior of AbstractSinglePropertyField does not automatically capture custom events unless explicitly instructed to do so. This is why I included the getElement().synchronizeProperty("value", "value-changed") call to ensure that the server-side state remains in sync with the client-side state whenever this particular event is fired. However, if this is not the case then I can remove this.

Comment (vursen) [reply articles/flow/integrations/react.adoc:271]: Could you try the example without it to see if it works? 

Based on what I see, AbstractSinglePropertyField not only updates the model value when the property changes but also performs some other logic like checking if the new value is valid:

https://github.com/vaadin/flow/blob/29e2e4adff260516c3d75789f52600d7e10543dd/flow-server/src/main/java/com/vaadin/flow/component/AbstractSinglePropertyField.java#L350-L354

File: articles/flow/integrations/react.adoc [hunk 2]
```diff
@@ -208,3 +208,78 @@ class ReactRouterLayoutElement extends ReactAdapterElement {
...
+=== Example Implementation:
+
+=== Create the Client-Side React Component
+First, define your React component. For this example, assume a simple text input component.
+[source,jsx]
+----
+// File: frontend/react-text-input.tsx
+
+import React, { useState } from 'react';
+import ReactDOM from 'react-dom';
```
Comment (caalador) [inline articles/flow/integrations/react.adoc:228]:
```suggestion

```

File: articles/flow/integrations/react.adoc [hunk 3]
```diff
@@ -208,3 +208,78 @@ class ReactRouterLayoutElement extends ReactAdapterElement {
...
+
+        const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+            setValue(event.target.value);
+            this.dispatchEvent(new CustomEvent('value-changed', { detail: { value: event.target.value } }));
+        };
+
+        return (
+            <div>
+                <input type="text" value={value} onChange={handleChange} />
+                <span>{value.length} characters</span>
```
Comment (caalador) [inline articles/flow/integrations/react.adoc:247]:
```suggestion
                <span>{value?.length} characters</span>
```
Error running if value is undefined/null so use conditional if value exists.

File: articles/flow/integrations/react.adoc [hunk 4]
```diff
@@ -208,3 +208,78 @@ class ReactRouterLayoutElement extends ReactAdapterElement {
...
+            </div>
+        );
+    }
+}
```
Comment (caalador) [inline articles/flow/integrations/react.adoc:251]:
```suggestion
}

customElements.define('react-text-input', ReactTextInputElement);
```
Define element to be able to use it.

**Review by caalador — CHANGES_REQUESTED** [review]

File: articles/flow/integrations/react.adoc [hunk 5]
```diff
@@ -208,3 +208,84 @@ class ReactRouterLayoutElement extends ReactAdapterElement {
 
 customElements.define('react-router-layout', ReactRouterLayoutElement);
 ----
+
+[[wrap-react-component]]
+== Wrapping a React component into Flow component
```
Comment (caalador) [inline articles/flow/integrations/react.adoc:213]:
```suggestion
== Wrapping a React Component Into a Flow Component
```
I'm not certain about these heading cases, but I guess something like this.

File: articles/flow/integrations/react.adoc [hunk 6]
```diff
@@ -208,3 +208,84 @@ class ReactRouterLayoutElement extends ReactAdapterElement {
...
+----
+// File: frontend/react-text-input.tsx
+
+import React, { useState } from 'react';
+import {ReactAdapterElement, type RenderHooks} from "Frontend/generated/flow/ReactAdapter";
+
+class ReactTextInputElement extends ReactAdapterElement {
+constructor() {
+super();
+}
```
Comment (caalador) [inline articles/flow/integrations/react.adoc:238]:
```suggestion
    constructor() {
        super();
    }
```
Minor formatting.

File: articles/flow/integrations/react.adoc [hunk 7]
```diff
@@ -208,3 +208,84 @@ class ReactRouterLayoutElement extends ReactAdapterElement {
...
+import com.vaadin.flow.component.dependency.JsModule;
+import com.vaadin.flow.component.AbstractSinglePropertyField;
+
+@Tag("react-text-input")
+@JsModule("./react-text-input.tsx")
+public class ReactTextField extends AbstractSinglePropertyField<ReactTextField, String> {
+public ReactTextField() {
+super("value", "", false);
+}
+}
```
Comment (caalador) [inline articles/flow/integrations/react.adoc:277]:
```suggestion
public class ReactTextField extends AbstractSinglePropertyField<ReactTextField, String> {
    public ReactTextField() {
        super("value", "", false);
    }
}
```
Formatting

**Review by caalador — APPROVED** [review]

File: articles/flow/integrations/react.adoc [hunk 8]
```diff
@@ -208,3 +208,84 @@ class ReactRouterLayoutElement extends ReactAdapterElement {
 
 customElements.define('react-router-layout', ReactRouterLayoutElement);
 ----
+
+[[wrap-react-component]]
+== Wrapping a React Component Into a Flow Component
```
Comment (mshabarov) [inline articles/flow/integrations/react.adoc:213]: This should be like "Wrapping a React Input Component Into a Flow Field Component".

Sorry for commenting on a closed PR, but just noticed one small remark. Could you change it in another PR ?

@AlainaFaisal

**Pull request merged**